cpp/oneapi/dal/algo/triangle_counting/backend/cpu/relabel_kernels.hpp

```diff
@@ -0,0 +1,118 @@
+    });
+    dal::detail::parallel_sort(degree_id_pairs, degree_id_pairs + vertex_count);
+}
+
+template <typename Cpu>
+void fill_new_degrees_and_ids_cpu(std::pair<std::int32_t, std::size_t>* degree_id_pairs,
+                                  std::int32_t* new_ids,
+                                  std::int32_t* degrees_relabel,
+                                  std::int64_t vertex_count) {
+    dal::detail::threader_for(vertex_count, vertex_count, [&](std::int32_t n) {
+        degrees_relabel[n] = degree_id_pairs[n].first;
+        new_ids[degree_id_pairs[n].second] = n;
+    });
+}
+
+template <typename Cpu>
+void parallel_prefix_sum_cpu(std::int32_t* degrees_relabel,
+                             std::int64_t* offsets,
+                             std::int64_t* part_prefix,
+                             std::int64_t* local_sums,
+                             size_t block_size,
+                             std::int64_t num_blocks,
+                             std::int64_t vertex_count) {
+    dal::detail::threader_for(num_blocks, num_blocks, [&](std::int64_t block) {
+        std::int64_t local_sum = 0;
+        std::int64_t block_end =
+            std::min((std::int64_t)((block + 1) * block_size), (std::int64_t)vertex_count);
+        for (std::int64_t i = block * block_size; i < block_end; i++) {
+            local_sum += degrees_relabel[i];
+        }
+        local_sums[block] = local_sum;
+    });
+
+    std::int64_t total = 0;
+    for (size_t block = 0; block < num_blocks; block++) {
+        part_prefix[block] = total;
+        total += local_sums[block];
+    }
+    part_prefix[num_blocks] = total;
+
+    dal::detail::threader_for(num_blocks, num_blocks, [&](std::int64_t block) {
+        std::int64_t local_total = part_prefix[block];
+        std::int64_t block_end =
+            std::min((std::int64_t)((block + 1) * block_size), (std::int64_t)vertex_count);
+        for (std::int64_t i = block * block_size; i < block_end; i++) {
+            offsets[i] = local_total;
+            local_total += degrees_relabel[i];
+        }
+    });
+
+    offsets[vertex_count] = part_prefix[num_blocks];
+}
+
+template <typename Cpu>
+void fill_relabeled_topology_cpu(const std::int32_t* vertex_neighbors,
+                                 const std::int64_t* edge_offsets,
+                                 std::int32_t* vertex_neighbors_relabel,
+                                 std::int64_t* edge_offsets_relabel,
+                                 std::int64_t* offsets,
+                                 std::int32_t* new_ids,
+                                 std::int64_t vertex_count) {
+    dal::detail::threader_for(vertex_count + 1, vertex_count + 1, [&](std::int32_t n) {
+        edge_offsets_relabel[n] = offsets[n];
+    });
+
+    dal::detail::threader_for(vertex_count, vertex_count, [&](std::int32_t u) {
+        for (const std::int32_t* v = vertex_neighbors + edge_offsets[u];
```

Comment:
v is pointer to int32_t - is this the place for some bugs when we switch to another data type in containers or so

Reply:
Updated to the correct version